localstack/services/ec2/resource_providers/aws_ec2_prefixlist.py

```diff
@@ -0,0 +1,135 @@
+# LocalStack Resource Provider Scaffolding v2
+from __future__ import annotations
+
+from pathlib import Path
+from typing import Optional, TypedDict
+
+import localstack.services.cloudformation.provider_utils as util
+from localstack.services.cloudformation.resource_provider import (
+    OperationStatus,
+    ProgressEvent,
+    ResourceProvider,
+    ResourceRequest,
+)
+
+
+class EC2PrefixListProperties(TypedDict):
+    AddressFamily: Optional[str]
+    MaxEntries: Optional[int]
+    PrefixListName: Optional[str]
+    Arn: Optional[str]
+    Entries: Optional[list[Entry]]
+    OwnerId: Optional[str]
+    PrefixListId: Optional[str]
+    Tags: Optional[list[Tag]]
+    Version: Optional[int]
+
+
+class Tag(TypedDict):
+    Key: Optional[str]
+    Value: Optional[str]
+
+
+class Entry(TypedDict):
+    Cidr: Optional[str]
+    Description: Optional[str]
+
+
+REPEATED_INVOCATION = "repeated_invocation"
+
+
+class EC2PrefixListProvider(ResourceProvider[EC2PrefixListProperties]):
+    TYPE = "AWS::EC2::PrefixList"  # Autogenerated. Don't change
+    SCHEMA = util.get_schema_path(Path(__file__))  # Autogenerated. Don't change
+
+    def create(
+        self,
+        request: ResourceRequest[EC2PrefixListProperties],
+    ) -> ProgressEvent[EC2PrefixListProperties]:
+        """
+        Create a new resource.
+
+        Primary identifier fields:
+          - /properties/PrefixListId
+
+        Required properties:
+          - PrefixListName
+          - MaxEntries
+          - AddressFamily
+
+
+
+        Read-only properties:
+          - /properties/PrefixListId
+          - /properties/OwnerId
+          - /properties/Version
+          - /properties/Arn
+
+        IAM permissions required:
+          - EC2:CreateManagedPrefixList
+          - EC2:DescribeManagedPrefixLists
+          - EC2:CreateTags
+
+        """
+        model = request.desired_state
+        create_params = util.select_attributes(
+            model, ["PrefixListName", "Entrie", "MaxEntries", "AddressFamily", "Tags"]
+        )
+
+        if "Tags" in create_params:
+            create_params["TagSpecifications"] = [
+                {"ResourceType": "prefix-list", "Tags": create_params.pop("Tags")}
+            ]
+
+        response = request.aws_client_factory.ec2.create_managed_prefix_list(**create_params)
+        model["Arn"] = response["PrefixList"]["PrefixListId"]
+        model["OwnerId"] = response["PrefixList"]["OwnerId"]
+        model["PrefixListId"] = response["PrefixList"]["PrefixListId"]
+        model["Version"] = response["PrefixList"]["Version"]
```

Comment:
In the future I think we should _always_ use the asynchronous form of this method (i.e. the default from the scaffolding) to make sure the resource provider ensures that resource has created correctly. It's not just about time taken to create the resource, but using a resource provider in CFn should be atomic i.e. when asked to create a resource, either the resource is created successfully or it fails. In resource providers we should make sure the resource has been created by using a `describe_*` or `get_*` method.

Not a blocker, but something to consider for the future.

cc @dominikschubert